### ISSUE TITLE
Unexpected results on sets of points

### ISSUE THREAD

**Issue by ismaelGomez87** [aleksandrbazhin/ellipsoid_fit_python#5]
Hi @aleksandrbazhin,

I am interested on using your code for some applications on molecular characterization. However, when I compute the ellipsoid for certain point sets I find unexpected results. See below the screenshots of the outcome.

![Set1](https://user-images.githubusercontent.com/56299750/71472337-ecf02d00-27d2-11ea-95f8-ec652b90cb98.png)
![Set2](https://user-images.githubusercontent.com/56299750/71472339-ef528700-27d2-11ea-8acc-c10f75d8b3f7.png)

I also find the following warning when running over the 1st set:

```
plot_ellipsoid.py:19: RuntimeWarning: invalid value encountered in double_scalars
  r = (a * b * c) ** (1. / 3.)
```

Could you please assist? I can share the data if needed.

Best,
Ismael.

**Comment by aleksandrbazhin** [comment]
Hi. 
Glad that the code is helpful. 
You can try disabling data regularization as it just throws out some of the points which is useful mainly for magnetometer calibrations. 
I will try to research the problems you are facing if you provide datasets which cause errors.

**Comment by aleksandrbazhin** [comment]
Also the code computes only the closest ellipsoid to the one with data points on its surface not enclosing all the points. Your data seem to be not suited for that and your results may as well be the closest to that kind of ellipsoids.

**Comment by aleksandrbazhin** [comment]
I think what you need is minimum value enclosing ellipsoid like https://github.com/minillinim/ellipsoid

**Comment by aleksandrbazhin** [comment]
Another option which may be good enough for classification is to construct a convex hull first, then compute an ellipsoid fitting it with this (my) repo.

**Comment by ismaelGomez87** [comment]
Hi @aleksandrbazhin,

thanks for your replies. What I need is to adjust an ellipsoid with minimum error to a set of points that are not necessarily placed in a 2D manifold (i.e. there may be no surface embedded in 3D that has all the points). If I understand correctly, your code wouldn't be suited for this task. Would the one you linked work for this?

I could construct the convex hull and select the points that are placed in the surface. Would that be enough to build an ellipsoid with your code?

Thanks,
Ismael.

**Comment by aleksandrbazhin** [comment]
Yes, the code I linked seems to do the job - find the minimal ellipsoid to contain all the points, but I did not look into the code. 
The convex hull method also should work. There is a built-in method into scipy package for convex hull construction. 
You should try both in my opinion and decide what works best. Or look for something else suited for your data.

**Comment by aleksandrbazhin** [comment]
Tried to describe my view on situation at #11
